extensions/ql-vscode/src/pure/variant-analysis-filter-sort.ts

```diff
@@ -13,6 +13,10 @@ export type RepositoriesFilterSortState = {
   sortKey: SortKey;
 }
 
+export type RepositoriesFilterSortStateWithIds = RepositoriesFilterSortState & {
+  repositoryIds?: number[];
+}
```

Comment:
Was there a reason we couldn't add this field to the original `RepositoriesFilterSortState` ? At the moment the extra type is making all the method definitions quite verbose while the actual implementation is short.

Reply:
We could it add it to the original type as well, but I think that would make it less clear that this property should not be used in the view. The view should explicitly not use the `repositoryIds` property because it would mean that once you select a repository using a checkbox, all other repositories would be hidden. By making this a separate type, we can ensure that the front-end doesn't use this field, rather than it being something implicit.

Reply:
Gotcha. 

> The view should explicitly not use the `repositoryIds` property because it would mean that once you select a repository using a checkbox, all other repositories would be hidden.

Can you expand on this a bit? Which view are we talking about?

Reply:
Sure! The view is the variant analysis view, which keeps track of the state [here](https://github.com/github/vscode-codeql/blob/e0230375e36ee23fd1a0378506ff212a09ccf86f/extensions/ql-vscode/src/view/variant-analysis/VariantAnalysis.tsx#L61-L62). As you can see, there are two different states we keep track of: the sort and filter state and the selected repositories. In the view, the selected repositories state is only used for keeping track of which repositories have their checkbox checked and for sending it to the extension.

If we combine the two types, there would also be a `repositoryIds` on the `filterSortState` which might be enticing to use in the view as well, instead of this separate state. However, this state is actually used for filtering which repositories are visible [here](https://github.com/github/vscode-codeql/blob/73ff057df00ed82d00556c74a139469add776723/extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisAnalyzedRepos.tsx#L58). So, if you start using this state for keeping track of selected checkboxes, you would find that once you select a checkbox, all other repositories disappear.

We can of course still combine the types, but it feels somewhat weird to me that there's a field on the type used by the view which actually "breaks" the selection if we started using it.

I was able to somewhat simplify the type checks so we now only have 1 method for checking the types, rather than two.